InformationScripting/src/parsing/QueryParser.h

```diff
@@ -60,10 +61,10 @@ class INFORMATIONSCRIPTING_API QueryParser
 		Type typeOf(const QString& text);
 		QPair<QStringList, QList<QChar> > split(const QString& text, const QList<QChar>& splitChars);
 
-		Query* parseQuery(const QString& text);
-		QList<Query*> parseList(const QString& text);
-		Query* parseOperator(const QString& text, bool connectInput = false);
-		QList<Query*> parseOperatorPart(const QString& text);
+		Query* parseQuery(Query* parent, const QString& text);
```

Comment:
In fact, compared to the current solution, this doesn't even present any more overhead. But can't we take this a bit further and only allow queries that actually care about the Executor to store it (and we won't need a virtual function).

My principle problem is not the overhead, but the fact that we force all queries to care about things which they shouldn't. If we can skip this, that would be better, but otherwise a solution like the current one is ok.

In your view, would you prefer a QE filed in each query, as opposed to the current solution?